src/mlpack/core/math/random.hpp

```diff
@@ -78,6 +78,38 @@ inline int RandInt(const int lo, const int hiExclusive)
 }
 
 /**
+ * Generates a random integer from range [0, N) with given probabilities
+ * P(0), P(1), ..., P(N-1) (arbitrary discrete distribution).
+ * 
+ * @param weights Probabilities of individual numbers. Note that the function
+ *                accepts unnormalized probabilties as long as they are
+ *                non-negative and sum to a positive number.
+ * @return A random integer sampled from the specified distribution.
+ */
```

Comment:
I think that instead of adding this overload of `RandInt()` we could use the functionality already available with `DiscreteDistribution`.  Instead of

```
x = RandInt(weights);
```

we can create a `DiscreteDistribution` and use it:

```
DiscreteDistribution d(weights);
x = d.Random();
```

What do you think?